### PR TITLE
Some nix adjustments 

### PR DESCRIPTION
**Pull request by MangoIV**
- add a formatter that can be run with `nix fmt`
- add more ghc versions to the output
- remove overrides for non-existent `flake-utils` input
- ~~fix eventual breakage due to the use of `name` instead of `pname`~~ This has been fixed upstream, so I changed it back to `name`
- update inputs

### CODE REVIEW

**Comment by MangoIV** [comment]
I think I'm missing some doc updates still, will add those later, if desired.

File: src/Cornelis/Utils.hs [hunk 1]
```diff
@@ -24,10 +24,6 @@ import           Neovim hiding (err)
 import           Neovim.API.Text
 import           Neovim.Context.Internal (Neovim(..), retypeConfig)
 
-
-deriving via (Ap (Neovim e) a) instance Semigroup a => Semigroup (Neovim e a)
```
Comment (isovector) [inline src/Cornelis/Utils.hs:28]: Bad merge?

Comment (MangoIV) [reply src/Cornelis/Utils.hs:28]: Those instances have been upstreamed d in neovim-hs, afaict, I got duplicate instances after nixpkgs update, maybe that's the wrong fix though, maybe there should be a constraint instead. Sorry I'll fix that.

Comment (isovector) [reply src/Cornelis/Utils.hs:28]: You're right, sorry! I forgot! Maybe we should update the `package.yaml` file to account for the new version of `neovim-hs.`

**Comment by isovector** [comment]
I've updated the package file, so I'm happy to merge this as soon as the conflicts are resolved.

**Comment by MangoIV** [comment]
I think this should be ready, thank you!

**Comment by isovector** [comment]
Thanks!

**Pull request merged**